plugins/woocommerce-beta-tester/api/tracks/class-tracks-debug-log.php

```diff
@@ -23,18 +23,32 @@ class Tracks_Debug_Log {
 	 */
 	private $source = 'tracks';
 
+	/**
+	 * Get the logger instance.
+	 * 
+	 * @return WC_Logger_Interface|null
+	 */
+	private function get_logger() {
+		// Sometimes between installations of WC versions the logger will not be available.
+		if ( ! $this->logger && function_exists( 'wc_get_logger' ) ) {
+			$this->logger = wc_get_logger();
+		}
+
+		return $this->logger;		
+	}
+
 	/**
 	 * Initialize hooks.
 	 */
 	public function __construct() {
-		include_once WC_ABSPATH . 'includes/tracks/class-wc-tracks-client.php';
-		include_once WC_ABSPATH . 'includes/tracks/class-wc-tracks-footer-pixel.php';
+		// WooCommerce might not be installed/activated between installs of WC versions.
+		if ( class_exists( 'WooCommerce' ) ) {
+			include_once WC_ABSPATH . 'includes/tracks/class-wc-tracks-client.php';
+			include_once WC_ABSPATH . 'includes/tracks/class-wc-tracks-footer-pixel.php';
 
-		$logger       = wc_get_logger();
-		$this->logger = $logger;
-
-		add_action( 'admin_footer', array( $this, 'log_footer_pixels' ), 5 );
-		add_action( 'pre_http_request', array( $this, 'log_remote_pixels' ), 10, 3 );
+			add_action( 'admin_footer', array( $this, 'log_footer_pixels' ), 5 );
```

Comment:
I guess I'm most concerned about this line, whether we will definitely get a new instance of this class constructed at some point between installs of WooCommerce or whether we could end up in a situation where this code is never loaded.

Reply:
Is your main concern around events between upgrades/installs?

Currently, the biggest risk I see is this code not being called because this constructor is not behind a hook and is just loaded in the order that WP loads these plugins (alphabetically).  If this plugin folder or the WC plugin folder were renamed to swap their order, this code would no longer run.

But this risk existed before and is limited to testing so I don't see it as a huge liability, unless there's something specific you have in mind.

Reply:
> Is your main concern around events between upgrades/installs?

yes that was my main concern, but as you say its probably not a big liability.

> Currently, the biggest risk I see is this code not being called because this constructor is not behind a hook and is just loaded in the order that WP loads these plugins (alphabetically). If this plugin folder or the WC plugin folder were renamed to swap their order, this code would no longer run.

Yikes! Maybe we can enhance to run this code via a hook in future.